compiler/passes/scopeResolve.cpp

```diff
@@ -1703,136 +1716,288 @@ static void    buildBreadthFirstModuleList(Vec<ModuleSymbol*>* modules,
                                            Vec<ModuleSymbol*>* current,
                                            Vec<ModuleSymbol*>* alreadySeen);
 
-static Symbol* lookup(BaseAST* scope, const char* name)
-{
-  Vec<Symbol*>  symbols;
+// Given a name and a scope, determine the symbol referred by that name in the
+// context of that scope.
+static Symbol* lookup(BaseAST* scope, const char* name) {
+  Symbol * symbolResult;
+  std::vector<Symbol * > symbolOptions;
   Vec<BaseAST*> nestedscopes;
-  lookup(scope, name, symbols, nestedscopes);
 
-  symbols.set_to_vec();
+  // Lydia note: I would like to investigate the effect of traversing all call
+  // expressions of the "." accessor.
 
-  if (symbols.n == 0)
-    // No symbols found.
+
+  // Call inner lookup on scope, the name, the symbols return vector, and the
+  // vector of ASTs already visited.
+  lookup(scope, name, symbolOptions, nestedscopes);
+
+  int numFound = symbolOptions.size();
+  if (numFound == 0) {
+    // No symbols found for this name
+      symbolResult = NULL;
+  } else if (numFound == 1) {
+    // A unique symbol found for this name
+      symbolResult = symbolOptions.front();
+  } else {
+    // Multiple symbols found for this name.  If at least one of them isn't
+    // a function, we need to handle it now so error out.  Otherwise, function
+    // resolution will handle it.
+    for_vector(Symbol, sym, symbolOptions) {
+      if (!isFnSymbol(sym))
+        USR_FATAL_CONT(sym, "Symbol %s multiply defined", name);
+    }
+    USR_STOP();
+    symbolResult = NULL;
+  }
+
+  return symbolResult;
+  // return symbol matching the name
+}
+
+// Determines and obtains a method by the given name on the given type
+//
+// This function uses the same methodology as isMethodName but returns the
+// symbol found instead of just a boolean
+static Symbol* getMethod(const char* name, Type* type) {
+  if (!strcmp(name, type->symbol->name))
```

Comment:
Just a thought: I tend to be cautious of writing !strcmp(x, y) rather than strcmp(x, y) == 0.  It so easy to
interpret '!' as "false" where the intent is "matched".